packages/client/src/MetricsPublisher.ts

```diff
@@ -79,8 +50,10 @@ export class MetricsPublisher {
     }
 
     private async publish(report: MetricsReport, streamId: string, partitionKey: string): Promise<void> {
+        await wait(Math.random() * this.config.maxPublishDelay)
         try {
             await this.publisher.publish(streamId, report, {
+                timestamp: report.period.end,
```

Comment:
Hmm.. I'd keep the `timestamp = Date.now()` to more accurately represent when the message was actually published. The "event logical timestamp" is already included in the payload?

Reply:
On the other hand, what is the point of having the feature which allows custom timestamp of a message if messages should always have `Date.now()` as the publish timestamp. I find it logical that timestamp represent the data (when it was collected), not when it was delivered.